src/pytest_workflow/plugin.py

```diff
@@ -36,35 +39,46 @@ def pytest_collect_file(path, parent):
 class YamlFile(pytest.File):
     """This class collects YAML files and turns them into test items."""
 
+    def __init__(self, path, parent):
+        super(YamlFile, self).__init__(path, parent=parent)
+
     def collect(self):
         with self.fspath.open() as yaml_file:
-            yaml_content = yaml.load(yaml_file)
-        validate_schema(yaml_content)
-        yield WorkflowItem(self.fspath.basename, self, yaml_content)
+            yaml_content = yaml.safe_load(yaml_file)
+        yield WorkflowTestsCollector(self.fspath.basename, self, yaml_content)
 
 
-class WorkflowItem(pytest.Item):
+class WorkflowTestsCollector(pytest.Collector):
     """This class defines a pytest item. That has methods for running tests."""
 
     def __init__(self, name, parent, yaml_content: dict):
+        validate_schema(yaml_content)
         self.yaml_content = yaml_content
+        self.name = name
```

Comment:
Isn't this handled in the `super` call already?

Reply:
tested it, and it does. Thanks